tools/wake/main.cpp

```diff
@@ -176,75 +165,61 @@ void inspect_database(const CommandLineOptions &clo, Database &db, const std::st
     return;
   }
 
-  std::unordered_map<long, JobReflection> captured_jobs = {};
-  Set<long> intersected_job_ids = suniversal<long>();
+  std::vector<std::vector<std::string>> collect_ands = {};
 
-  if (!clo.job_ids.empty()) {
-    intersected_job_ids =
-        apply_inspection_query(captured_jobs, std::move(intersected_job_ids), clo.job_ids,
-                               [&](auto a) { return db.job_ids_matching(a); });
-  }
+  // Process --job
+  make_and_group(clo.job_ids, "cast(job_id as TEXT)", "", collect_ands);
```

Comment:
It just turns out that all these command line options happen to be `vector<vector<string>>`? What is the second layer of `vector` for in something like `--job`?

Reply:
> It just turns out  ...

Nope, was intentionally designed that way!

so all inspection flags that take params have the following format

`--flag 'foo,bar' --flag 'baz,bat'`

which gets parsed into `[[foo, bar], [baz, bat]]` The inner vectors represent `or` ops while the outer vector is one big `and` op

Reply:
in `--job`s case, `--job '1?,2?' --job '?5'` is perfectly valid to say jobs 15 and 25

Reply:
Is this and/or documented in the usage text?